cumulusci/core/utils.py

```diff
@@ -151,15 +155,123 @@ def decode_to_unicode(content):
 
 
 def merge_config(configs):
-    """recursively deep-merge the configs into one another (highest priority comes first)"""
-    new_config = {}
+    """
+    First remove any flow steps that are being overridden so that there are no conflicts
+    or different step types after merging. Then recursively deep-merge the configs into
+    one another (highest priority comes first)
+    """
+    configs = cleanup_flow_step_override_conflicts(configs)
 
+    new_config = {}
     for name, config in configs.items():
         new_config = dictmerge(new_config, config, name)
 
     return new_config
 
 
+def cleanup_flow_step_override_conflicts(configs: T.List[dict]) -> T.List[dict]:
+    """
+    If a flow step is been overridden with a step of a different type (i.e. tas, flow),
+    then we need to set the step that is _lower_ in precedence order to an empty dict ({}).
+    If we don't, then we will end up with both "flow" and "task" listed in the step_config which
+    leads to an error when CumulusCI attempts to run that step in the flow.
+
+    We need to also account for scenarios where a single flow step
+    is being overriden more than once.
+
+    Example:
+    A flow step in the universal config is being overridden somewhere
+    in the project_config, which is being overridden by additional_yaml.
+
+    The while loop below is how we account for these scenarios.
+    """
+    config_precedence_order = [
+        "additional_yaml",
+        "project_local_config",
+        "project_config",
+        "global_config",
+        "universal_config",
+    ]
```

Comment:
Why are you using a hardcoded list here and not simply processing all configs in the order they are listed? `merge_config` specifies that the configs are in priority order (highest precedence first).